### PR TITLE
Fixes #2962 - Clean up fields parameter choices in GET APIs

### LINKED ISSUE #2962: Clean up fields parameter choices in GET APIs
Entities have two types of fields - `attributes` and `relationships`. The `attributes` are stored as part of the JSON document in the backend database for an entity. The `relationship` fields are not stored as part of the JSON document. 

When an entity is retrieved in GET operation (either list entities or get a single entity):
- Setting `attributes` in an entity does not require additional database lookup and incurs no extra cost other than making the response payload larger. Hence, the `fields` param used for controlling the payload size typically does not include `attribute` field names. The exception to this rule *maybe* some large attributes of type array.
- Setting `relationships` of an entity however requires a lookup into the `entity_relationship` table and hence has database lookup cost. Hence, the `fields` param typically includes `relationship` field names.

This *backward-incompatible* change prunes the `fields` param list of attribute field names and will include only `relationship` field names.

### PR DESCRIPTION
**Pull request by sureshms**
### Describe your changes :

See #2962 for more details

#
### Type of change :

- [x] Bug fix
- [x] Improvement
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by harshach — APPROVED** [review]

File: catalog-rest-service/src/test/java/org/openmetadata/catalog/resources/teams/UserResourceTest.java [hunk 1]
```diff
@@ -13,6 +13,7 @@
 
 package org.openmetadata.catalog.resources.teams;
 
+import static java.util.List.*;
```
Comment (mithmatt) [inline catalog-rest-service/src/test/java/org/openmetadata/catalog/resources/teams/UserResourceTest.java:16]: Wildcard import..

File: catalog-rest-service/src/test/java/org/openmetadata/catalog/resources/teams/UserResourceTest.java [hunk 2]
```diff
@@ -650,11 +641,13 @@ public void validateCreatedEntity(User user, CreateUser createRequest, Map<Strin
     for (UUID roleId : Optional.ofNullable(createRequest.getRoles()).orElse(Collections.emptyList())) {
       expectedRoles.add(new EntityReference().withId(roleId).withType(Entity.ROLE));
     }
-    if (!expectedRoles.isEmpty()) {
-      assertEquals(expectedRoles.size(), user.getRoles().size());
-      for (EntityReference role : expectedRoles) {
-        TestUtils.existsInEntityReferenceList(user.getRoles(), role.getId(), true);
-      }
+    System.out.println("XXX expectedRoles " + expectedRoles);
```
Comment (mithmatt) [inline catalog-rest-service/src/test/java/org/openmetadata/catalog/resources/teams/UserResourceTest.java:644]: Remove print statements?

File: catalog-rest-service/src/test/java/org/openmetadata/catalog/resources/pipelines/PipelineResourceTest.java [hunk 3]
```diff
@@ -522,22 +516,34 @@ void delete_nonEmptyPipeline_4xx() {
...
     pipeline =
         byName
             ? getPipelineByName(pipeline.getFullyQualifiedName(), fields, ADMIN_AUTH_HEADERS)
             : getPipeline(pipeline.getId(), fields, ADMIN_AUTH_HEADERS);
-    assertListNotNull(pipeline.getOwner(), pipeline.getService(), pipeline.getServiceType(), pipeline.getTasks());
+    assertListNotNull(pipeline.getService(), pipeline.getServiceType());
+    // Some fields are not set and hence are null - tested elsewhere
+    assertListNotNull(
+        pipeline.getOwner(),
+        /*pipeline.getTasks(),
```
Comment (mithmatt) [inline catalog-rest-service/src/test/java/org/openmetadata/catalog/resources/pipelines/PipelineResourceTest.java:542]: Remove commented code?

**Review by mithmatt — APPROVED** [review]

File: catalog-rest-service/src/test/java/org/openmetadata/catalog/resources/pipelines/PipelineResourceTest.java (hunk 3 above)
Comment (sureshms) [reply catalog-rest-service/src/test/java/org/openmetadata/catalog/resources/pipelines/PipelineResourceTest.java:542]: That was to call out, we are not specifically testing those fields... :)

**Pull request merged**